contracts/libs/trader-pool/TraderPoolInvest.sol

```diff
@@ -90,21 +100,84 @@ library TraderPoolInvest {
         }
     }
 
+    function investTokens(
+        ITraderPool.PoolParameters storage poolParameters,
+        mapping(address => mapping(uint256 => uint256)) storage investsInBlocks,
+        EnumerableSet.AddressSet storage positions,
+        uint256[] calldata amounts,
+        address[] calldata tokens
+    ) external returns (uint256 toMintLP) {
+        address baseToken = poolParameters.baseToken;
+
+        TraderPool traderPool = TraderPool(address(this));
+
+        (uint256 totalBase, , , ) = poolParameters.getNormalizedPoolPriceAndPositions();
+
+        for (uint256 i; i < tokens.length; i++) {
+            require(
+                !traderPool.coreProperties().isBlacklistedToken(tokens[i]),
+                "TP: token in blacklist"
+            );
+            uint256 baseAmount;
+            IERC20Metadata(tokens[i]).safeTransferFrom(
+                msg.sender,
+                address(this),
+                amounts[i].to18(IERC20Metadata(tokens[i]).decimals())
+            );
+
+            if (tokens[i] != baseToken) {
+                (baseAmount, ) = traderPool.priceFeed().getNormalizedPriceOut(
+                    tokens[i],
+                    baseToken,
+                    amounts[i]
+                );
+
+                if (positions.contains(tokens[i])) {
+                    emit ActivePortfolioExchanged(baseToken, tokens[i], baseAmount, amounts[i]);
+                } else {
+                    positions.add(tokens[i]);
+                    emit Exchanged(msg.sender, baseToken, tokens[i], baseAmount, amounts[i]);
+                }
+            } else {
+                baseAmount = amounts[i];
+            }
+
+            toMintLP += _calculateToMintLP(poolParameters, investsInBlocks, totalBase, baseAmount);
+        }
+
+        traderPool.updateTo(msg.sender, toMintLP, toMintLP);
```

Comment:
```suggestion
        traderPool.updateTo(msg.sender, toMintLP, <???>);
```

You have to calculate the total invested amount and pass it as a third parameter